app/src/main/java/org/xbmc/kore/ui/sections/video/MovieListFragment.java

```diff
@@ -425,5 +426,18 @@ public void bindView(Cursor cursor) {
                 artView.setTransitionName("a" + dataHolder.getId());
             }
         }
+
+        private String getMetaInfo(Cursor cursor) {
+            int runtime = cursor.getInt(MovieListQuery.RUNTIME) / 60;
+            int movieYear = cursor.getInt(MovieListQuery.YEAR);
+
+            ArrayList<String> duration = new ArrayList<>();
+            if (runtime > 0)
+                duration.add(String.format(context.getString(R.string.minutes_abbrev),
+                                           String.valueOf(runtime)));
+            duration.add(String.valueOf(movieYear));
+
+            return TextUtils.join(" | ", duration);
+        }
```

Comment:
I have no problem with this, but i must point out that the previous version was slightly more efficient, in terms of not needing as many objects allocated (the arraylist). This code is run for every list item, and even though i don't think this constitutes a problem, this kind of micro-optimizations might add up over time.

Anyway, i'm probably splitting hairs here.

Reply:
If scrolling becomes slow we will need to refactor this completely and generate the content, at least the heavy parts, on a separate thread.